respokeSDK/src/main/java/com/digium/respokesdk/RespokeClient.java

```diff
@@ -513,6 +533,187 @@ public RespokeGroup getGroup(String groupIDToFind) {
+     * other method signature that allows `maxMessages` to be specified.
+     *
+     * @param groupIds The groups to pull history for
+     * @param completionListener The callback called when this async operation has completed
+     */
+    public void getGroupHistories(ArrayList<String> groupIds,
+                                  final GroupHistoriesCompletionListener completionListener) {
+        getGroupHistories(groupIds, 1, completionListener);
+    }
+
+    /**
+     * Retrieve the history of messages that have been persisted for 1 or more groups. Only those
+     * messages that have been marked to be persisted when sent will show up in the history.
+     *
+     * @param groupIds The groups to pull history for
+     * @param maxMessages The maximum number of messages per group to pull. Must be >= 1
+     * @param completionListener The callback called when this async operation has completed
+     */
+    public void getGroupHistories(final ArrayList<String> groupIds, final Integer maxMessages,
+                                  final GroupHistoriesCompletionListener completionListener) {
+        if (!isConnected()) {
+            getGroupHistoriesError(completionListener, "Can't complete request when not connected, " +
+                    "Please reconnect!");
+            return;
+        }
+
+        if ((maxMessages == null) || (maxMessages < 1)) {
+            getGroupHistoriesError(completionListener, "maxMessages must be at least 1");
+            return;
+        }
+
+        if ((groupIds == null) || (groupIds.size() == 0)) {
+            getGroupHistoriesError(completionListener, "At least 1 group must be specified");
+            return;
+        }
+
+        Uri.Builder builder = new Uri.Builder();
+        builder.appendQueryParameter("limit", maxMessages.toString());
+        for (int i = 0; i < groupIds.size(); i++) {
+            builder.appendQueryParameter("groupIds", groupIds.get(i));
+        }
+
+        String urlEndpoint = "/v1/group-histories" + builder.build().toString();
+        signalingChannel.sendRESTMessage("get", urlEndpoint, null,
+                new RespokeSignalingChannel.RESTListener() {
+            @Override
+            public void onSuccess(Object response) {
+                if (!(response instanceof JSONObject)) {
+                    getGroupHistoriesError(completionListener, "Invalid response from server");
+                    return;
+                }
+
+                final JSONObject json = (JSONObject) response;
+                final HashMap<String, ArrayList<RespokeGroupMessage>> results = new HashMap<>();
+
+                for (Iterator<String> keys = json.keys(); keys.hasNext();) {
+                    final String key = keys.next();
+
+                    try {
+                        final JSONArray jsonMessages = json.getJSONArray(key);
+
+                        final ArrayList<RespokeGroupMessage> messageList = new ArrayList<>();
```

Comment:
Thanks for bringing this up. I looked into it some and it looks like it only creates new arrays when adding additional items exceeds the initial capacity of the array. Additionally, it looks like LinkedList tends to be a poor choice in the general case, since it has a high memory footprint due to the the overhead of having many objects + pointers vs. a single array. 

I'm going to leave this as an ArrayList for now but I appreciate the feedback. If you feel very strongly about it being a LinkedList I'd be willing to discuss it more.

http://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist